backend/zoho/zoho.go

```diff
@@ -0,0 +1,1240 @@
+	params := url.Values{}
+	params.Set("filename", name)
+	params.Set("parent_id", parent)
+	params.Set("override-name-exist", strconv.FormatBool(true))
+	formReader, contentType, overhead, err := rest.MultipartUpload(in, nil, "content", name)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to make multipart upload")
+	}
+
+	contentLength := overhead + size
+	opts := rest.Opts{
+		Method:           "POST",
+		Path:             "/upload",
+		Body:             formReader,
+		ContentType:      contentType,
+		ContentLength:    &contentLength,
+		Options:          options,
+		Parameters:       params,
+		TransferEncoding: []string{"identity"},
+	}
+
+	var resp *http.Response
+	var uploadResponse *api.UploadResponse
+	err = f.pacer.CallNoRetry(func() (bool, error) {
+		resp, err = f.srv.CallJSON(ctx, &opts, nil, &uploadResponse)
+		return shouldRetry(resp, err)
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "upload error")
+	}
+	if len(uploadResponse.Uploads) != 1 {
+		return nil, errors.New("upload: invalid response")
+	}
+	// Received meta data is missing size so we have to read it again.
+	info, err := f.readMetaDataForID(ctx, uploadResponse.Uploads[0].Attributes.RessourceID)
+	if err != nil {
+		return nil, err
+	}
+
+	return info, nil
+}
+
+// PutUnchecked the object into the container
+//
+// This will produce an error if the object already exists
+//
+// Copy the reader in to the new object which is returned
+//
+// The new object may have been created if an error is returned
+func (f *Fs) PutUnchecked(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (fs.Object, error) {
+	size := src.Size()
+	remote := src.Remote()
+
+	// Create the directory for the object if it doesn't exist
+	leaf, directoryID, err := f.dirCache.FindPath(ctx, remote, true)
+	if err != nil {
+		return nil, err
+	}
+
+	if isSimpleName(leaf) {
+		info, err := f.upload(ctx, f.opt.Enc.FromStandardName(leaf), directoryID, size, in, options...)
+		if err != nil {
+			return nil, err
+		}
+		return f.newObjectWithInfo(ctx, remote, info)
+	}
+
+	tempName := "rcloneTemp" + random.String(8)
+	info, err := f.upload(ctx, tempName, directoryID, size, in, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	o, err := f.newObjectWithInfo(ctx, remote, info)
+	if err != nil {
+		return nil, err
+	}
+	return o, o.(*Object).rename(ctx, leaf)
+}
+
+// Mkdir creates the container if it doesn't exist
+func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+	_, err := f.dirCache.FindDir(ctx, dir, true)
+	return err
+}
+
+// deleteObject removes an object by ID
+func (f *Fs) deleteObject(ctx context.Context, id string) (err error) {
+	var resp *http.Response
+	opts := rest.Opts{
+		Method:       "PATCH",
+		Path:         "/files",
+		ExtraHeaders: map[string]string{"Accept": "application/vnd.api+json"},
+	}
+	delete := api.WriteMultiMetadataRequest{
+		Meta: []api.WriteMetadata{
+			{
+				Attributes: api.WriteAttributes{
+					Status: "51",
```

Comment:
51?

Reply:
Status 51 is deleted. I guess i could make this a named constant but's it's the only status used and it's only used in this one place so I didn't bother.

Reply:
It just seemed such a random number! Make a const if you want but if it is just used here then a comment would be fine.